.github/workflows/ci.yml

```diff
@@ -1,32 +1,109 @@
 name: CI
 
-env:
-  COMPOSE_INTERACTIVE_NO_CLI: 1
-  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
 on:
   push:
-    branches: [ master ]
   pull_request:
-    branches: [ master ]
+    branches: [ 3.x, 2.x, master ]
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
 jobs:
   tests:
     runs-on: ubuntu-latest
+    container: abrardev/tenancy:latest
 
     strategy:
       matrix:
-        php: ["8.1"]
-        laravel: ["^9.0"]
+        php: ['8.1']
+        laravel: ['^8.0', '^9.0']
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: root
+          POSTGRES_DB: main
+        ports:
+          - 5432/tcp
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 3
+
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: main
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+      mysql2:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: main
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+      mssql:
+        image: mcr.microsoft.com/mssql/server:2019-latest
+        ports:
+          - 1433/tcp
+        env:
+          ACCEPT_EULA: Y
+          SA_PASSWORD: P@ssword
+        options: --health-cmd "echo quit | /opt/mssql-tools/bin/sqlcmd -S 127.0.0.1 -l 1 -U sa -P P@ssword"
+
+      redis:
+        image: redis
+        ports:
+          - 6379/tcp
+        options: --health-cmd="redis-cli ping" --health-interval=10s --health-timeout=5s --health-retries=3
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Start docker containers
-      run: PHP_VERSION=${{ matrix.php }} docker-compose up -d
-    - name: Install dependencies
-      run: docker-compose exec -T test composer require --no-interaction "laravel/framework:${{ matrix.laravel }}"
-    - name: Run tests
-      run: ./test
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, sqlsrv, pdo_sqlsrv, redis, bcmath, soap, intl, gd, exif, iconv, imagick, fileinfo
+          coverage: pcov
```

Comment:
Can't this be done directly inside the docker container? That's sort of the point of this PR, that we will be using a Docker container made exactly for our CI. We shouldn't be reconfiguring it in a separate step, it should be already fully configured in the Dockerfile.

Reply:
You are right. We shouldn't need these extensions here when they are already enabled in Dockerfile.